server/go/subscription/manager.go

```diff
@@ -0,0 +1,67 @@
+package subscription
+
+import "sync"
+
+type Manager struct {
+	subscriptions map[string][]Subscriber
```

Comment:
Isn't something like `topics` a better match than subscriptions?

Reply:
I'm not sure what you mean, but this `observer` patter is pretty standard for this kind of things. If I understand you correctly, `topics` would mean to have different queues, much like in Kafka. That is a lot more complex, but in essence it's just a complicated observer